hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkTable.java

```diff
@@ -91,7 +91,7 @@ protected HoodieIndex getIndex(HoodieWriteConfig config, HoodieEngineContext con
   protected Option<HoodieTableMetadataWriter> getMetadataWriter(
       String triggeringInstantTimestamp,
       HoodieFailedWritesCleaningPolicy failedWritesCleaningPolicy) {
-    if (config.isMetadataTableEnabled() || metaClient.getTableConfig().isMetadataTableAvailable()) {
+    if (!HoodieTableMetadata.isMetadataTable(metaClient.getBasePathV2().toString()) && config.isMetadataTableEnabled()) {
       // if any partition is deleted, we need to reload the metadata table writer so that new table configs are picked up
```

Comment:
I guess the check `HoodieTableMetadata.isMetadataTable(metaClient.getBasePathV2().toString())` is unnecessary because MDT already disable its MDT explicitly.

and we should also fix: https://github.com/apache/hudi/blob/418a47e6eec8bd5cb06d17e11d28c3e90cf8455d/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java#L175